lib/index.ts

```diff
@@ -54,43 +57,64 @@ function parseFromFile(
 }
 
 export function buildDependencyTree(
-  /* manifestFile: PaketManifest, */ lockFile: PaketLock, includeDev: boolean = false,
+  manifestFile: PaketDependencies, lockFile: PaketLock, includeDev: boolean = false,
 ) {
   const depTree = {
     dependencies: {},
     name: '',
     version: '',
   } as DepTree;
 
+  const dependencies: { [dep: string]: {} } = {};
+
+  for (const group of manifestFile) {
+    for (const dep of group.dependencies) {
+      const current = dep as NugetDependency;
+      if (current.name) {
+        dependencies[current.name] = {
+          name: current.name,
+        };
+      }
+    }
+  }
+
   for (const group of lockFile.groups) {
     const isDev = group.name === 'build' || group.name === 'test' || group.name === 'tests';
     if (isDev && !includeDev) {
       continue;
     }
 
     for (const dep of group.dependencies) {
-      depTree.dependencies[dep.name] = {
-        depType: isDev ? DepType.dev : DepType.prod,
-        dependencies: buildSubTree(dep.dependencies),
-        name: dep.name,
-        version: dep.version,
-      };
+      if (dependencies[dep.name]) {
+        dependencies[dep.name] = {
+          depType: isDev ? DepType.dev : DepType.prod,
+          dependencies: buildSubTree(dep.dependencies, group.dependencies, isDev),
+          name: dep.name,
+          version: dep.version,
+        };
+      }
```

Comment:
Quickest way to do this. Revisiting now to address like nuget parser https://github.com/snyk/dotnet-deps/blob/develop/src/lib/nuget-dep-flat-list.ts#L180 and https://github.com/snyk/dotnet-deps/blob/develop/src/lib/nuget-deptree-builder.ts#L42